ggshield/cmd/iac/scan/diff.py

```diff
@@ -194,7 +196,7 @@ def iac_scan_diff(
         ).get_http_headers(),
     )
 
-    if not scan.success or not isinstance(scan, IaCDiffScanResult):
+    if not isinstance(scan, IaCDiffScanResult):
```

Comment:
I'm not sure to understand why you removed this condition here? Same in `all.py`.

Reply:
That is required to fix pyright error: it thinks it is possible for both `scan.success` to be False and `scan` to be an instance of `IaCDiffScanResult`. If this were to happen, then the code in the `if` block would be wrong because it would pass an `IaCDiffScanResult` to `handle_scan_errors()`.

I can see how the change is not intuitive, but the alternative requires adding some `assert`, which is not super nice either:

```python
if not scan.success:
    assert isinstance(scan, Detail)
    handle_scan_error(client, scan)
    return None
assert isinstance(scan, IaCDiffScanResult)
return scan
```

The second `assert` is required because pyright does not know that if we did not enter the `if` block then `scan` is always an instance of `IaCDiffScanResult`.